graphics/src/AssimpLoader.cc

```diff
@@ -0,0 +1,733 @@
+        this->GenerateTextureName(_scene, assimpMat, "Metalness"),
+        metalTexture);
+    pbr.SetRoughnessMap(
+        this->GenerateTextureName(_scene, assimpMat, "Roughness"),
+        roughTexture);
+  }
+  else
+  {
+    // Load the textures separately
+    ret = assimpMat->GetTexture(aiTextureType_METALNESS, 0, &texturePath);
+    if (ret == AI_SUCCESS)
+    {
+      auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+          this->GenerateTextureName(_scene, assimpMat, "Metalness"));
+      pbr.SetMetalnessMap(texName, texData);
+    }
+    ret = assimpMat->GetTexture(
+        aiTextureType_DIFFUSE_ROUGHNESS, 0, &texturePath);
+    if (ret == AI_SUCCESS)
+    {
+      auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+          this->GenerateTextureName(_scene, assimpMat, "Roughness"));
+      pbr.SetRoughnessMap(texName, texData);
+    }
+  }
+#endif
+  ret = assimpMat->GetTexture(aiTextureType_NORMALS, 0, &texturePath);
+  if (ret == AI_SUCCESS)
+  {
+    auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+        this->GenerateTextureName(_scene, assimpMat, "Normal"));
+    // TODO(luca) different normal map spaces
+    pbr.SetNormalMap(texName, NormalMapSpace::TANGENT, texData);
+  }
+  ret = assimpMat->GetTexture(aiTextureType_EMISSIVE, 0, &texturePath);
+  if (ret == AI_SUCCESS)
+  {
+    auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+        this->GenerateTextureName(_scene, assimpMat, "Emissive"));
+    pbr.SetEmissiveMap(texName, texData);
+  }
+  unsigned int uvIdx = 0;
+  ret = assimpMat->GetTexture(
+      aiTextureType_LIGHTMAP, 0, &texturePath, NULL, &uvIdx);
+  if (ret == AI_SUCCESS)
+  {
+    auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+        this->GenerateTextureName(_scene, assimpMat, "Lightmap"));
+    pbr.SetLightMap(texName, uvIdx, texData);
+  }
+#ifndef GZ_ASSIMP_PRE_5_2_0
+  double value;
+  ret = assimpMat->Get(AI_MATKEY_METALLIC_FACTOR, value);
+  if (ret == AI_SUCCESS)
+  {
+    pbr.SetMetalness(value);
+  }
+  ret = assimpMat->Get(AI_MATKEY_ROUGHNESS_FACTOR, value);
+  if (ret == AI_SUCCESS)
+  {
+    pbr.SetRoughness(value);
+  }
+#endif
+  mat->SetPbrMaterial(pbr);
+  return mat;
+}
+
+//////////////////////////////////////////////////
+std::pair<std::string, ImagePtr> AssimpLoader::Implementation::LoadTexture(
+    const aiScene* _scene,
+    const aiString& _texturePath,
+    const std::string& _textureName) const
+{
+  std::pair<std::string, ImagePtr> ret;
+  // Check if the texture is embedded or not
+  auto embeddedTexture = _scene->GetEmbeddedTexture(_texturePath.C_Str());
+  if (embeddedTexture)
+  {
+    // Load embedded texture
+    ret.first = _textureName;
+    ret.second = this->LoadEmbeddedTexture(embeddedTexture);
+  }
+  else
+  {
+    ret.first = ToString(_texturePath);
+  }
+  return ret;
+}
+
+std::pair<ImagePtr, ImagePtr>
+    AssimpLoader::Implementation::SplitMetallicRoughnessMap(
```

Comment:
Nit: This appears to be completely uncovered by tests.  If it is unused, can we remove it, otherwise, I would appreciate a follow-up to correct that.

Reply:
I think this is due to Codecov running on Ubuntu 20.04 rather than Ubuntu 22.04, specifically:

* The function is called [here](https://github.com/gazebosim/gz-common/blob/2c0773159e0299d0ed506b1a0f6ee541cb0bdd44/graphics/src/AssimpLoader.cc#L398), which is under an `#ifndef` block that doesn't get compiled in Assimp < 5.2.x (Ubuntu 20.04). The function populates metallic and roughness maps.
* The unit test for this functionality is [here](https://github.com/gazebosim/gz-common/blob/2c0773159e0299d0ed506b1a0f6ee541cb0bdd44/graphics/src/AssimpLoader_TEST.cc#L682-L686), I tested it locally on my Ubuntu 22.04 machine and indeed had to fix a few issues when I ran it, but again this test is `#ifndef`d away on Ubuntu 20.04 since the old assimp library doesn't support metallic roughness maps and it would fail. It runs on github actions for Jammy but not on Focal and is not picked up by codecov.

An example of an action from some time ago while I was iterating is [here](https://github.com/gazebosim/gz-common/actions/runs/2747512033), specifically Focal passes because the block is not compiled but Jammy fails [here](https://github.com/gazebosim/gz-common/runs/7542761772?check_suite_focus=true#step:4:4663) because of an issue that I fixed since.

If we want to increase coverage I can put the whole code block that is effectively dead in a `#ifndef` block as well, otherwise I don't know if there is a way (or it is desirable) to have codecov run on a newer version of Ubuntu.

That being said there is a small code path that is never tested, which is assets (such as gltf) that have an external texture, I'll add this in an upcoming PR.
This is specifically the path going through [this](https://app.codecov.io/gh/gazebosim/gz-common/compare/393/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr%20comments&utm_term=gazebosim#D4L363) and [this](https://app.codecov.io/gh/gazebosim/gz-common/compare/393/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr%20comments&utm_term=gazebosim#D4L484) line.